### PR TITLE
[TS][Inverisify] Adding support for RxJS 6 

### PR DESCRIPTION
**Pull request by gualtierim**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Adding support to rxjs6 and patch http method for typescript-inverisify's language

### CODE REVIEW

**Comment by gualtierim** [comment]
@nicokoenig @TiFu @taxpon @sebastianhaas @macjohnny

File: samples/client/petstore/typescript-inversify/api/pet.service.ts [hunk 1]
```diff
@@ -11,9 +11,9 @@
  */
 /* tslint:disable:no-unused-variable member-ordering */
 
-import { Observable } from "rxjs/Observable";
-import 'rxjs/add/operator/map';
-import 'rxjs/add/operator/toPromise';
+import { Observable } from 'rxjs/Observable';
+
+import { map } from 'rxjs/operators';
```
Comment (macjohnny) [inline samples/client/petstore/typescript-inversify/api/pet.service.ts]: I think `map` should be added to the reserved keywords

File: modules/openapi-generator/src/main/resources/typescript-inversify/HttpClient.mustache [hunk 2]
```diff
@@ -1,28 +1,39 @@
 import IHttpClient from "./IHttpClient";
-import { Observable } from "rxjs/Observable";
+
+{{^useRxJS6}}
+import { Observable } from 'rxjs/Observable';
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-inversify/HttpClient.mustache]: i suggest to use double quotes as in the rest of the code

File: modules/openapi-generator/src/main/resources/typescript-angular/package.mustache [hunk 3]
```diff
@@ -28,7 +28,12 @@
     "@angular/compiler": "^{{ngVersion}}",
     "core-js": "^2.4.0",
     "reflect-metadata": "^0.1.3",
-    "rxjs": "^{{rxjsVersion}}"
+    {{^useRxJS6}}
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-angular/package.mustache]: why is this changed? it is not related to typescript-inversify

Comment (gualtierim) [reply modules/openapi-generator/src/main/resources/typescript-angular/package.mustache]: I added a config's property`useRxJS6` to allow the user to decide the using of rxjs6 or rxjs5

File: samples/client/petstore/typescript-inversify/api/pet.service.ts (hunk 1 above)
Comment (gualtierim) [reply samples/client/petstore/typescript-inversify/api/pet.service.ts]: I used the standard import of map operator of rxjs as suggested in the documentation.

File: modules/openapi-generator/src/main/resources/typescript-angular/package.mustache (hunk 3 above)
Comment (macjohnny) [reply modules/openapi-generator/src/main/resources/typescript-angular/package.mustache]: here you are modifying the `typescript-angular` file

File: samples/client/petstore/typescript-inversify/api/pet.service.ts (hunk 1 above)
Comment (macjohnny) [reply samples/client/petstore/typescript-inversify/api/pet.service.ts]: yes, but if a user defines an api parameter named `map`, this clashes, so we should add `map` to the list of reserved keywords, similar to this:
https://github.com/OpenAPITools/openapi-generator/blob/50fddc701eabeb3025cc4df25212b9b757caeb40/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java#L64-L69

File: modules/openapi-generator/src/main/resources/typescript-angular/package.mustache (hunk 3 above)
Comment (gualtierim) [reply modules/openapi-generator/src/main/resources/typescript-angular/package.mustache]: Sorry, I did a big mistake

File: samples/client/petstore/typescript-inversify/api/pet.service.ts (hunk 1 above)
Comment (gualtierim) [reply samples/client/petstore/typescript-inversify/api/pet.service.ts]: I thought you refered to the reserved key keywords of javascript and I could not understand you. Thanks to the example. I added map to reserved keywords.

Comment (wing328) [reply samples/client/petstore/typescript-inversify/api/pet.service.ts]: I've filed https://github.com/OpenAPITools/openapi-generator/pull/3020 to rename "map" to "GlobalImportOperators" so that `map` doesn't need to be a reserved word.

**Comment by wing328** [comment]
@gualtierim thanks again for your contribution, which has been included in the v4.0.1 release (https://twitter.com/oas_generator/status/1135534738658062336)
